plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSnapshots.java

```diff
@@ -36,6 +36,22 @@ protected QueryRunner createQueryRunner()
         return createIcebergQueryRunner();
     }
 
+    @Test
+    public void testSnapshotContainsQueryId() {
+        String tableName = "test_snapshot_summary" + randomTableSuffix();
+        assertUpdate(format("CREATE TABLE %s (a bigint, b bigint)", tableName));
+        assertQuery(format(
+                "SELECT count(distinct json_extract_scalar(CAST(summary AS JSON), '$.presto_query_id')) from " +
+                        "\"%s$snapshots\"",
+                tableName),"VALUES(1)");
+        assertUpdate(format("INSERT INTO %s VALUES(1, 1)", tableName), 1);
+        assertQuery(format(
+            "SELECT count(distinct json_extract_scalar(CAST(summary AS JSON), '$.presto_query_id')) from " +
+                    "\"%s$snapshots\"",
+                tableName),"VALUES(2)");
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
```

Comment:
I'm working on generalizing this, but I haven't found a good way to know the query ID before hand.

I tried calling `getSession()` and then passing that session to `assertUpdate` and `assertQuery`, but the `preesto_query_id` didn't match (not sure if this is a sign of a bug in my implemtation).

As far as correctness goes, when looking at the snapshot summary, the query id seemed to be somewhat incremental between two consecutive writes.